source/AMDSClient.cpp

```diff
@@ -0,0 +1,347 @@
+			}
+			for(int y = 0, ySize = bufferGroupInfos.count(); y < ySize; y++){
+				if(clientIntrospectionRequest->bufferName() == "All")
+					bufferNameComboBox_->addItem(bufferGroupInfos.at(y).name());
+				qDebug() << bufferGroupInfos.at(y).name() << bufferGroupInfos.at(y).description() << bufferGroupInfos.at(y).units() << bufferGroupInfos.at(y).rank() << bufferGroupInfos.at(y).size().toString();
+				for(int x = 0, size = bufferGroupInfos.at(y).axes().count(); x < size; x++){
+					qDebug() << "\tAxis info at " << x << bufferGroupInfos.at(y).axes().at(x).name() << bufferGroupInfos.at(y).axes().at(x).description() << bufferGroupInfos.at(y).axes().at(x).units() << bufferGroupInfos.at(y).axes().at(x).size() << bufferGroupInfos.at(y).axes().at(x).isUniform() << bufferGroupInfos.at(y).axes().at(x).start() << bufferGroupInfos.at(y).axes().at(x).increment();
+				}
+			}
+		}
+	}
+	else if(clientRequest->requestType() == AMDSClientRequestDefinitions::Statistics){
+		AMDSClientStatisticsRequest *clientStatisticsRequest = qobject_cast<AMDSClientStatisticsRequest*>(clientRequest);
+		if(clientStatisticsRequest){
+			for(int x = 0, size = clientStatisticsRequest->packetStats().count(); x < size; x++)
+				qDebug() << "Packet Stats " << clientStatisticsRequest->packetStats().at(x).name() << ": " << clientStatisticsRequest->packetStats().at(x).allStats();
+		}
+	}
+	else if(clientRequest->requestType() == AMDSClientRequestDefinitions::StartTimePlusCount){
+		AMDSClientStartTimePlusCountDataRequest *clientStartTimePlusCountDataRequest = qobject_cast<AMDSClientStartTimePlusCountDataRequest*>(clientRequest);
+		if(clientStartTimePlusCountDataRequest){
+			for(int x = 0, size = clientStartTimePlusCountDataRequest->data().count(); x < size; x++){
+				AMDSFlatArray oneFlatArray = AMDSFlatArray(clientStartTimePlusCountDataRequest->uniformDataType(), clientStartTimePlusCountDataRequest->bufferGroupInfo().spanSize());
+				clientStartTimePlusCountDataRequest->data().at(x)->data(&oneFlatArray);
+
+//				qDebug() << "Data at " << x << oneFlatArray.vectorDouble();
+				qDebug() << "\nData at " << x << oneFlatArray.printData();
+			}
+		}
+	}
+	else if(clientRequest->requestType() == AMDSClientRequestDefinitions::RelativeCountPlusCount){
+		AMDSClientRelativeCountPlusCountDataRequest *clientRelativePlusCountDataRequest = qobject_cast<AMDSClientRelativeCountPlusCountDataRequest*>(clientRequest);
+		if(clientRelativePlusCountDataRequest){
+			for(int x = 0, size = clientRelativePlusCountDataRequest->data().count(); x < size; x++){
+				AMDSFlatArray oneFlatArray = AMDSFlatArray(clientRelativePlusCountDataRequest->uniformDataType(), clientRelativePlusCountDataRequest->bufferGroupInfo().spanSize());
+				clientRelativePlusCountDataRequest->data().at(x)->data(&oneFlatArray);
+
+//				qDebug() << "Data at " << x << oneFlatArray.vectorDouble();
+				qDebug() << "\nData at " << x << oneFlatArray.printData();
+			}
+		}
+	}
+
+	getFortuneButton->setEnabled(true);
```

Comment:
Not important at the moment, but this probably should be changed to something other than the "Get Fortune" Qt example ;)  Maybe something more specific to what we're doing ;D